app/src/main/java/com/bogdan/codeforceswatcher/features/problems/redux/reducers/ProblemsReducer.kt

```diff
@@ -9,8 +11,35 @@ fun problemsReducer(action: Action, state: AppState): ProblemsState {
     var newState = state.problems
 
     when (action) {
+        is ProblemsRequests.FetchProblems -> {
+            newState = newState.copy(status = ProblemsState.Status.PENDING)
+        }
+
         is ProblemsRequests.FetchProblems.Success -> {
-            newState = newState.copy(problems = action.problems)
+            newState = newState.copy(
+                problems = newState.problems.plus(action.problems),
```

Comment:
Maybe this is why UI is lagging ? ) shouldn't it be `problems = action.problems` ?